src/main/java/org/synyx/urlaubsverwaltung/account/AccountFormValidator.java

```diff
@@ -66,30 +68,55 @@ void validatePeriod(AccountForm form, Errors errors) {
         final LocalDate holidaysAccountValidFrom = form.getHolidaysAccountValidFrom();
         final LocalDate holidaysAccountValidTo = form.getHolidaysAccountValidTo();
 
-        validateDateNotNull(holidaysAccountValidFrom, "holidaysAccountValidFrom", errors);
-        validateDateNotNull(holidaysAccountValidTo, "holidaysAccountValidTo", errors);
+        validateDateNotNull(holidaysAccountValidFrom, ATTR_HOLIDAYS_ACCOUNT_VALID_FROM, errors);
+        validateDateNotNull(holidaysAccountValidTo, ATTR_HOLIDAYS_ACCOUNT_VALID_TO, errors);
 
         if (holidaysAccountValidFrom != null && holidaysAccountValidTo != null) {
-            boolean periodIsNotWithinOneYear = holidaysAccountValidFrom.getYear() != form.getHolidaysAccountYear()
-                || holidaysAccountValidTo.getYear() != form.getHolidaysAccountYear();
+            final int year = form.getHolidaysAccountYear();
+            final int fromYear = holidaysAccountValidFrom.getYear();
+            final int toYear = holidaysAccountValidTo.getYear();
+            final boolean invalidYear = fromYear != year || toYear != year;
+
+            if (invalidYear) {
+                if (fromYear != year) {
+                    reject(errors, ATTR_HOLIDAYS_ACCOUNT_VALID_FROM, msg("holidaysAccountValidFrom.invalidYear"), String.valueOf(year));
+                }
+                if (toYear != year) {
+                    reject(errors, ATTR_HOLIDAYS_ACCOUNT_VALID_TO, msg("holidaysAccountValidTo.invalidYear"), String.valueOf(year));
+                }
+                return;
+            }
+
             boolean periodIsOnlyOneDay = holidaysAccountValidFrom.equals(holidaysAccountValidTo);
-            boolean beginOfPeriodIsAfterEndOfPeriod = holidaysAccountValidFrom.isAfter(holidaysAccountValidTo);
+            if (periodIsOnlyOneDay) {
```

Comment:
`periodIsOnlyOneDay` können wir mit `beginOfPeriodIsAfterEndOfPeriod` drehen oder? Dann wird erst geschaut ob die Period korrekt in die Zukunft gerichtet ist und danach kann man dann ohne den return die Länge testen.

Reply:
bzw ich würde alle returns hier rausnehmen da die Fehlernachrichten gestacked werden und somit der Benutzer direkt alle Informationen bekommt und nicht eine nach der anderen.